### PR TITLE
Link order notes to transaction details

### LINKED ISSUE #1307: Link order note to transaction details page
Change payment intent id to a link to shortcut full order notes update mentioned in #1054

### PR DESCRIPTION
**Pull request by vbelolapotkov**
Fixes #1307 

#### Changes proposed in this Pull Request

* Make payment intent id in order notes a link pointing to transaction details page.
* Fix rendering transaction summary on details page for authorized but not captured transactions.

#### Testing instructions

* Create various transaction according to changes in `class-wc-payment-gateway-wcpay.php`.
* Check that order notes has link to transactions details page. Clicking the link should open the transaction details page.
* Test that authorized but not captured transaction details page renders without errors.

No new tests added.

-------------------

- [x] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by vbelolapotkov** [comment]
@luizreis fyi, I wasn't able to test all scenarios. Only with normal capture and manual capture are tested, so I'd really appreciate your help with testing if you have time for that. If no time for testing, it's totally fine, please let me know and I'll ask somebody else for assistance.

File: client/utils/charge/index.js [hunk 1]
```diff
@@ -66,13 +66,21 @@ export const getChargeStatus = ( charge = {} ) => {
...
+				refunded: 0,
+				net: 0,
+		  }
+		: {
+				currency: charge.currency,
+				amount: charge.amount,
+				fee: charge.application_fee_amount,
+				refunded: 0,
+				net: 0,
+		  };
```
Comment (luizreis) [inline client/utils/charge/index.js:83]: Works well! 🚀 🏆

**Review by luizreis — COMMENTED** [review]
@vbelolapotkov - sorry for the delay in reviewing this. I wasn't able to successfully test the `requires_action` scenario and left a comment with more details.

Overall the code looks good. But I'd like to know your opinion on whether we should create a new method to handle the order note creation. It looks like the code is pretty much the same across all statuses, with the only difference being the actual status. Do you think it's worth it? It might make translations a bit more strict but would make updates such as this one way easier.

File: includes/class-wc-payment-gateway-wcpay.php [hunk 2]
```diff
@@ -671,15 +673,17 @@ public function process_payment_for_order( $cart, $payment_information ) {
 					break;
 				case 'requires_action':
```
[the remark this answers is not in the document]
Comment (vbelolapotkov) [reply includes/class-wc-payment-gateway-wcpay.php]: Yes, you're right. I've tested with SCA card as well by cancelling the popup. After exploring Stripe logs it's clear we won't have charge_id at this step. So I removed the changes for this case in https://github.com/Automattic/woocommerce-payments/pull/1309/commits/0ff477ec18b1133a06da81d84bfd50288332963b

**Comment by vbelolapotkov** [comment]
Thanks for reviewing it @luizreis 

> But I'd like to know your opinion on whether we should create a new method to handle the order note creation. It looks like the code is pretty much the same across all statuses, with the only difference being the actual status. Do you think it's worth it? It might make translations a bit more strict but would make updates such as this one way easier.

I was thinking about it but realised there are too many different notes created in other cases. Also the `intent_id` is available and used only in a few notes. On the other hand I fully agree making this kind of changes to notes is really hard and some general refactoring to the way notes added to the order or payments processed are needed, but I don't see a clear path forward so I'd postpone refactoring for now.

**Comment by jrodger** [comment]
The class we're making changes to here is...large. We have made some steps to start refactoring functionality out, so that's good. We do need to tackle this more general issue at some point, but I do think it's outside the scope of this PR (and we're not adding many new lines here).

I've tested with a range of cards. The results match the existing link in the order details sub-heading in all cases. I've also tested the transaction detail screen for un-captured charges. Both look good 👍 

🚢

**Pull request merged**